spec.html

```diff
@@ -890,6 +890,30 @@ <h1>AsyncContext.Snapshot.prototype</h1>
       </emu-clause>
     </emu-clause>
 
+    <emu-clause id="sec-function-properties-of-the-asynccontext-snapshot-object">
+      <h1>Function Properties of the AsyncContext.Snapshot Object</h1>
+
+      <emu-clause id="sec-asynccontext-wrap">
+        <h1>AsyncContext.Snapshot.wrap ( _fn_ )</h1>
+        <p>This function returns a new function which restores the current value of all AsyncContext.Variable values when being invoked.</p>
+
+        <emu-alg>
+          1. If IsCallable(_fn_) is false, throw a TypeError exception.
+          1. Let _snapshot_ be AsyncContextSnapshot().
+          1. Let _closure_ be a new Abstract Closure with parameters (..._args_) that captures _snapshot_ and performs the following steps when called:
+            1. Let _thisArgument_ be the *this* value.
+            1. Let _previousContextMapping_ be AsyncContextSwap(_snapshot_).
+            1. Let _result_ be Completion(Call(_fn_, _thisArgument_, _args_)).
+            1. AsyncContextSwap(_previousContextMapping_).
+            1. Return _result_.
+          1. Let _length_ be ? LengthOfArrayLike(_fn_).
+          1. Let _name_ be ? Get(_fn_, *"name"*).
+          1. If _name_ is not a String, set _name_ to the empty String.
```

Comment:
Should the name of the wrapped function be `"wrapped " + name`?

Reply:
Done.

Reply:
In what sense is this done?